src/test/java/net/testudobank/tests/MvcControllerIntegTest.java

```diff
@@ -591,6 +596,201 @@ public void testReversalOfSimpleWithdraw() throws SQLException, ScriptException,
+   * $1. 
+   * 
+   * The sender's balance should decrease by the transfer amount.
+   */
+
+  @Test
+  public void testTransferPaysOffOverdraftBalance() throws SQLException, ScriptException { 
+    
+   //Initialize customer1 with a balance of $1000. Balance will be represented as pennies in DB.
+   double CUSTOMER1_BALANCE = 1000;
+   int CUSTOMER1_BALANCE_IN_PENNIES = MvcControllerIntegTestHelpers.convertDollarsToPennies(CUSTOMER1_BALANCE);
+   MvcControllerIntegTestHelpers.addCustomerToDB(dbDelegate, CUSTOMER1_ID, CUSTOMER1_PASSWORD, CUSTOMER1_FIRST_NAME, CUSTOMER1_LAST_NAME, CUSTOMER1_BALANCE_IN_PENNIES);
+
+   //Initialize customer2 with a balance of $100. Balance will be represented as pennies in DB. 
+   double CUSTOMER2_BALANCE = 100;
+   int CUSTOMER2_BALANCE_IN_PENNIES = MvcControllerIntegTestHelpers.convertDollarsToPennies(CUSTOMER2_BALANCE);
+   MvcControllerIntegTestHelpers.addCustomerToDB(dbDelegate, CUSTOMER2_ID, CUSTOMER2_PASSWORD, CUSTOMER2_FIRST_NAME, CUSTOMER2_LAST_NAME, CUSTOMER2_BALANCE_IN_PENNIES);
+
+   //Amount to transfer
+   int TRANSFER_AMOUNT = 100;
+   int TRANSFER_AMOUNT_IN_PENNIES = MvcControllerIntegTestHelpers.convertDollarsToPennies(TRANSFER_AMOUNT);
+
+
+   //Initializing users for the transfer
+   User CUSTOMER1 = new User();
+   User CUSTOMER2 = new User();
+   CUSTOMER1.setUsername(CUSTOMER1_ID);
+   CUSTOMER1.setPassword(CUSTOMER1_PASSWORD);
+
+   double CUSTOMER2_AMOUNT_TO_WITHDRAW = 201;
+   int CUSTOMER2_AMOUNT_TO_WITHDRAW_IN_PENNIES = MvcControllerIntegTestHelpers.convertDollarsToPennies(CUSTOMER2_AMOUNT_TO_WITHDRAW);
+   int CUSTOMER2_EXPECTED_OVERDRAFT_BALANCE = MvcControllerIntegTestHelpers.applyOverdraftInterest(CUSTOMER2_BALANCE_IN_PENNIES - CUSTOMER2_AMOUNT_TO_WITHDRAW_IN_PENNIES) * -1;
+   CUSTOMER2.setUsername(CUSTOMER2_ID);
+   CUSTOMER2.setPassword(CUSTOMER2_PASSWORD);
+   CUSTOMER2.setAmountToWithdraw(CUSTOMER2_AMOUNT_TO_WITHDRAW);
+
+   CUSTOMER1.setWhoToTransfer(CUSTOMER2_ID);
+   CUSTOMER1.setAmountToTransfer(TRANSFER_AMOUNT);
+
+    //Withdraw $201 from Customer2's account to cause an overdraft of $101.
+    controller.submitWithdraw(CUSTOMER2);
+
+    //Send the transfer request.
+    String returnedPage = controller.submitTransfer(CUSTOMER1);
+
+    //fetch customers table data from DB
+    List<Map<String,Object>> customer1SqlResult = jdbcTemplate.queryForList(String.format("SELECT * FROM Customers WHERE CustomerID='%s';", CUSTOMER1_ID));
+    Map<String, Object> customer1Data = customer1SqlResult.get(0);
+
+    List<Map<String,Object>> customer2SqlResult = jdbcTemplate.queryForList(String.format("SELECT * FROM Customers WHERE CustomerID='%s';", CUSTOMER2_ID));
+    Map<String, Object> customer2Data = customer2SqlResult.get(0);
+
+    //Verify that customer1's balance decreased by $100. 
+    assertEquals((CUSTOMER1_BALANCE_IN_PENNIES - TRANSFER_AMOUNT_IN_PENNIES), (int)customer1Data.get("Balance"));
+
+    //Verify that customer2's overdraft balance decreased by $100.
```

Comment:
Also verify that customer2's main balance is still $0, which is what makes this different from your next integ test below.